solr/bin/solr

```diff
@@ -873,6 +873,12 @@ if [[ "$SCRIPT_CMD" == "delete" ]]; then
   exit $?
 fi
 
+# Prevent any zk subcommands from going through with out invoking zk command
+if [[ "$SCRIPT_CMD" == "upconfig" || $SCRIPT_CMD == "downconfig" || $SCRIPT_CMD == "cp" || $SCRIPT_CMD == "rm" || $SCRIPT_CMD == "mv" || $SCRIPT_CMD == "ls" || $SCRIPT_CMD == "mkroot" ]]; then
```

Comment:
Feels a bit of duplication, for every sub command of any command we need to add it here to forbid it?
Cannot this be handled in SolrCLI, or is the main-command handled by shell and sub-commands by Java?

Reply:
Yeah, I did not like it.  main is shell and sub commands are java...  so thats why `bin/solr cp` actually is executed in java code.   Now, if we move this all over to more how the `bin/solr package <subcommand>` then we wouldn't need this.